feat: add new mode `weak_except_iboutlets` (#4975)
realm/SwiftLint: CHANGELOG.md

  `unallowed_symbols_severity`. It accepts the two values `warning` and `error`
  (default) as usual.  

>   [SimplyDanny](https://github.com/SimplyDanny)
> * Extend `implicitly_unwrapped_optional` rule with a new mode `weak_except_iboutlets`
>   that only checks `weak` variable with force-unwrapping operator.
>   [Ricky Tan](https://github.com/rickytan)

Reviewer: ```suggestion
  [SimplyDanny](https://github.com/SimplyDanny)

* Extend `implicitly_unwrapped_optional` rule with the new mode
  `weak_except_iboutlets` that only checks `weak` variables.  
  [Ricky Tan](https://github.com/rickytan)
```
